internal/analytics/deferred.go

```diff
@@ -18,10 +26,53 @@ type deferredData struct {
 }
 
 const deferredCfgKey = "deferrer_analytics"
+const deferrerFileName = "deferrer"
+
+func deferrerTimeStamp(cfg Configurable) string {
+	return filepath.Join(cfg.ConfigPath(), deferrerFileName)
+}
+
+func deferrerTimeStampDayOld(cfg Configurable) bool {
+	df := deferrerTimeStamp(cfg)
+	stat, err := os.Stat(df)
+	if err != nil {
+		logging.Errorf("Could not stat file: %s, error: %v", df, err)
+		return false
+	}
+	diff := time.Now().Sub(stat.ModTime())
+	return diff > 24*time.Hour
+}
+
+func runNonDeferredStateToolCommand(cfg Configurable) error {
+	exe, err := os.Executable()
+	if err != nil {
+		logging.Errorf("Could not determine State Tool executable: %v", err)
+		exe = "state"
+	}
+	cmd := exec.Command(exe, "--version")
+	cmd.SysProcAttr = osutils.SysProcAttrForNewProcessGroup()
+	cmd.Env = append(os.Environ(), fmt.Sprintf("%s=true", constants.DisableUpdates))
```

Comment:
Do we not need to route `stdout` and `stderr` here to avoid it printing the results of `state --version` to the user's terminal?

Reply:
They are set to nil by default, which should means `/dev/null`, `/dev/zero`...  But I think I will make that explicit, just to make sure...